### PR TITLE
Compare decrypt

### PR DESCRIPTION
**Pull request by xrwang**
decrypt `oldParameters` before comparing to a masterConfig

@rclark would love a review, I think I have set this up badly and that's causing testing issues...

### CODE REVIEW

File: lib/commands.js [hunk 1]
```diff
@@ -292,7 +294,24 @@ var operations = module.exports.operations = {
...
+              region: context.overrides.templateOptions.AWS_DEFAULT_REGION,
+              maxRetries: 10
+            });
+            var valueToDecrypt = context.oldParameters[key].replace(/^secure:/,'');
+            kms.decrypt({
+              CiphertextBlob: new Buffer(valueToDecrypt, 'base64')
+            }, function (err, data) {
+              if (err) return context.abort();
+              var decryptedOldParams = new Buffer(data.Plaintext, 'base64').toString('utf-8');
+              var decryptedMaster = masterConfigJSON[key]+'new';
```
Comment (rclark) [inline lib/commands.js]: Where does the `new` come from?

File: test/commands.test.js [hunk 2]
```diff
@@ -621,7 +622,12 @@ test('[commands.operations.getMasterConfig] success', function(assert) {
     }
   });
 
-  context.oldParameters = { old: 'stale' };
+  AWS.stub('KMS', 'decrypt', function(params, data) {
+    return 'data';
+  });
+
+  context.oldParameters = { old: 'secure:staleelats' };
+  context.overrides = { templateOptions : { AWS_DEFAULT_REGION : 'us-east-1'}};
```
Comment (rclark) [inline test/commands.test.js]: Does this PR introduce a dependency on an `$AWS_DEFAULT_REGION` env var that wasn't there before? Or is it just this test that needed to be changed?

File: test/commands.test.js [hunk 3]
```diff
@@ -621,7 +622,12 @@ test('[commands.operations.getMasterConfig] success', function(assert) {
     }
   });
 
-  context.oldParameters = { old: 'stale' };
+  AWS.stub('KMS', 'decrypt', function(params, data) {
+    return 'data';
+  });
+
+  context.oldParameters = { old: 'secure:staleelats' };
```
Comment (rclark) [inline test/commands.test.js]: where can I get some stale elats?

image

File: lib/commands.js (hunk 1 above)
Comment (xrwang) [inline lib/commands.js]: goops that was a left over from testing. removed!

File: test/commands.test.js (hunk 2 above)
Comment (xrwang) [inline test/commands.test.js]: Good question -- the `AWS_DEFAULT_REGION` comes from `templateOptions.AWS_DEFAULT_REGION`, but I'm not sure if this change is not generalizable enough for cfn-config.

Comment (xrwang) [inline test/commands.test.js]: by `generalizable enough for cfn-config`, I mean if it's something that we can count on as always being there.

File: lib/commands.js [hunk 4]
```diff
@@ -292,7 +294,23 @@ var operations = module.exports.operations = {
       if (err) return context.abort();
       Object.keys(masterConfigJSON).forEach(function(key) {
         if (context.oldParameters.hasOwnProperty(key)) {
-          context.oldParameters[key] = masterConfigJSON[key];
+          if (context.oldParameters[key].indexOf('secure') > -1) {
+            var kms = new AWS.KMS({
+              region: context.overrides.templateOptions.AWS_DEFAULT_REGION,
```
Comment (rclark) [inline lib/commands.js]: Yeah this is definitely dubious. The presence of `.overrides.templateOptions` even existing depends on whether or not you passed it in with `overrides` [when calling commands.update() in the first place](https://github.com/mapbox/cfn-config/blob/be0f679075e6526073db2b4bc2caa58baf10fcd4/lib/commands.js#L113). 

Instead, there should be [a `config` object in scope](https://github.com/mapbox/cfn-config/blob/be0f679075e6526073db2b4bc2caa58baf10fcd4/lib/commands.js#L54) with a `.region` property that indicates the region in which the update is being performed.

**Pull request merged**